tests/flow/test_v7_encode_decode.py

```diff
@@ -0,0 +1,137 @@
+        query = "MATCH ()-[e]->() WITH e ORDER by id(e) return COLLECT(id(e))"
+        expected_full_graph_nodes_id = redis_graph.query(query)
+        # Delete 3 edges.
+        redis_graph.query("MATCH ()-[e]->() WHERE id(e) IN [7,14,20] DELETE e")
+        expected_nodes_id_after_delete = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        # Validate no compaction, all IDs are the same
+        self.env.assertEquals(expected_nodes_id_after_delete.result_set, actual.result_set)
+        # Validate reuse of edges ids - create 3 edges.
+        redis_graph.query("UNWIND range (0,2) as i CREATE ()-[:R]->()")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected_full_graph_nodes_id.result_set, actual.result_set)
+
+    def test05_multiple_edges_over_multiple_keys(self):
+        graph_name = "multiple_edges_over_multiple_keys"
+        redis_graph = Graph(graph_name, redis_con)
+        # Create 3 edges meta keys
+        redis_graph.query("CREATE (n1 {val:1}), (n2 {val:2}) WITH n1, n2 UNWIND range(0,20) as i CREATE (n1)-[:R {val:i}]->(n2)")
+        # Return all the edges, before and after saving & loading the RDB, and check equality
+        query = "MATCH ()-[e]->() return e"
+        expected = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected.result_set, actual.result_set)
+
+    def test06_no_compaction_on_multiple_edges_delete(self):
+        graph_name = "no_compaction_on_multiple_edges_delete"
+        redis_graph = Graph(graph_name, redis_con)
+        # Create 3 nodes meta keys
+        redis_graph.query("CREATE (n1 {val:1}), (n2 {val:2}) WITH n1, n2 UNWIND range(0,20) as i CREATE (n1)-[:R]->(n2)")
+        # Return all the edges, before and after saving & loading the RDB, and check equality
+        query = "MATCH ()-[e]->() WITH e ORDER by id(e) return COLLECT(id(e))"
+        expected_full_graph_nodes_id = redis_graph.query(query)
+        # Delete 3 edges.
+        redis_graph.query("MATCH ()-[e]->() WHERE id(e) IN [7,14,20] DELETE e")
+        expected_nodes_id_after_delete = redis_graph.query(query)
+        # Save RDB & Load from RDB
+        redis_con.execute_command("DEBUG", "RELOAD")
+        actual = redis_graph.query(query)
+        # Validate no compaction, all IDs are the same
+        self.env.assertEquals(expected_nodes_id_after_delete.result_set, actual.result_set)
+        # Validate reuse of edges ids - create 3 edges.
+        redis_graph.query("MATCH (n1 {val:1}), (n2 {val:2}) WITH n1, n2 UNWIND range (0,2) as i CREATE ()-[:R]->()")
+        actual = redis_graph.query(query)
+        self.env.assertEquals(expected_full_graph_nodes_id.result_set, actual.result_set)
+
+    def test07_index_after_encode_decode_in_v7(self):
```

Comment:
Would it matter if we would have data in the `index_after_encode_decode_in_v7` graph?
As at the moment the graph contains a single index but has no entities.

Reply:
Those tests are meant to validate each of the following values encoding- decoding flow

1. nodes
2. deleted nodes
3. edges - normal and edges array
4. deleted edges - normal and edges array
5. index